File: docs/clause-and-effect/in-prolog/worksheet-04-acyclic-directed-graph.pl
a(g, h).
a(g, d).
a(e, d).
a(h, f).
a(e, f).
a(a, e).
a(a, b).
a(b, f).
a(b, c).
a(f, c).

path(X, X).
path(X, Y) :- a(X, Z), path(Z, V).

%% path(f, f).
%% path(a, c).
%% path(g, e).
%% path(g, X).
%% path(X, h).

File: docs/old/clause-and-effect/worksheet-04-acyclic-directed-graph.js
let logic = require ("logic-db")

let a = new logic.db_t

a.i (["g", "h"])
a.i (["g", "d"])
a.i (["e", "d"])
a.i (["h", "f"])
a.i (["e", "f"])
a.i (["a", "e"])
a.i (["a", "b"])
a.i (["b", "f"])
a.i (["b", "c"])
a.i (["f", "c"])

let path = new logic.db_t

path.i (["?x", "?x"])
path.i (["?x", "?y"]) .cond ((the) => {
    let z = new logic.var_t
    return a.o ([the.var.x, z]) .and (path.o ([z, the.var.y]))
})

path.query_log (100) (["g", "?x"])
path.query_log (100) (["?x", "h"])
